src/Npgsql/PostgresDatabaseInfo.cs

```diff
@@ -140,45 +137,58 @@ static string GenerateLoadTypesQuery(bool withRange, bool withMultirange, bool l
-    (typtype = 'p' AND typname IN ('record', 'void')) OR -- Some special supported pseudo-types
-    (typtype = 'a' AND (  -- Array of...
-        elemtyptype IN ('b', 'r', 'm', 'e', 'd') OR -- Array of base, range, multirange, enum, domain
-        (elemtyptype = 'p' AND elemtypname IN ('record', 'void')) OR -- Arrays of special supported pseudo-types
-        (elemtyptype = 'c' AND {(loadTableComposites ? "ns.nspname NOT IN ('pg_catalog', 'information_schema', 'pg_toast')" : "elemrelkind='c'")}) -- Array of user-defined free-standing composites (not table composites) by default
+    typtype IN ('b', 'r', 'm', 'e', 'd') OR
+    (typtype = 'c' AND {(loadTableComposites ? "ns.nspname NOT IN ('pg_catalog', 'information_schema', 'pg_toast')" : "relkind='c'")}) OR
+    (typtype = 'p' AND typname IN ('record', 'void')) OR
+    (typtype = 'a' AND (
+        elemtyptype IN ('b', 'r', 'm', 'e', 'd') OR
+        (elemtyptype = 'p' AND elemtypname IN ('record', 'void')) OR
+        (elemtyptype = 'c' AND {(loadTableComposites ? "ns.nspname NOT IN ('pg_catalog', 'information_schema', 'pg_toast')" : "elemrelkind='c'")})
     ))
 ORDER BY CASE
-       WHEN typtype IN ('b', 'e', 'p') THEN 0           -- First base types, enums, pseudo-types
-       WHEN typtype = 'r' THEN 1                        -- Ranges after
-       WHEN typtype = 'm' THEN 2                        -- Multiranges after
-       WHEN typtype = 'c' THEN 3                        -- Composites after
-       WHEN typtype = 'd' AND elemtyptype <> 'a' THEN 4 -- Domains over non-arrays after
-       WHEN typtype = 'a' THEN 5                        -- Arrays after
-       WHEN typtype = 'd' AND elemtyptype = 'a' THEN 6  -- Domains over arrays last
-END;";
+       WHEN typtype IN ('b', 'e', 'p') THEN 0
+       WHEN typtype = 'r' THEN 1
+       WHEN typtype = 'm' THEN 2
+       WHEN typtype = 'c' THEN 3
+       WHEN typtype = 'd' AND elemtyptype <> 'a' THEN 4
+       WHEN typtype = 'a' THEN 5
+       WHEN typtype = 'd' AND elemtyptype = 'a' THEN 6
+END;"
+#if NET6_0_OR_GREATER
+    .ReplaceLineEndings("\n");
+#else
+    .Replace("\r\n", "\n").Replace('\r', '\n');
+#endif
 
     static string GenerateLoadCompositeTypesQuery(bool loadTableComposites)
         => $@"
--- Load field definitions for (free-standing) composite types
 SELECT typ.oid, att.attname, att.atttypid
 FROM pg_type AS typ
 JOIN pg_namespace AS ns ON (ns.oid = typ.typnamespace)
 JOIN pg_class AS cls ON (cls.oid = typ.typrelid)
 JOIN pg_attribute AS att ON (att.attrelid = typ.typrelid)
 WHERE
   (typ.typtype = 'c' AND {(loadTableComposites ? "ns.nspname NOT IN ('pg_catalog', 'information_schema', 'pg_toast')" : "cls.relkind='c'")}) AND
-  attnum > 0 AND     -- Don't load system attributes
+  attnum > 0 AND
   NOT attisdropped
-ORDER BY typ.oid, att.attnum;";
+ORDER BY typ.oid, att.attnum;"
+#if NET6_0_OR_GREATER
```

Comment:
Any particular value in having #if here? I'd rather we kept things simple and just did the version which works everywhere.

Reply:
Another thought is to keep the comments and just strip them out here, same as we clean up the newlines - it would help keep some explanations for what's happening above.

Reply:
> Any particular value in having #if here? I'd rather we kept things simple and just did the version which works everywhere.

No it's just a bit sad that the version which works everywhere needs to process the string twice but I don't really mind removing the `#if` part.

> Another thought is to keep the comments and just strip them out here, same as we clean up the newlines - it would help keep some explanations for what's happening above.

Yeah I also found it a bit sad to rip out the comment because I find them really helpful in some instances but since you said you don't mind, I didn't bother.
The only way that comes to my mind to strip out comments would be a regex or some kind of hand-written minimalistic parser. Do you have another idea and if not, do you really think that's worthwhile and which way would you prefer then?

Reply:
> No it's just a bit sad that the version which works everywhere needs to process the string twice but I don't really mind removing the #if part.

Yeah, this is code that runs once in the application lifetime, so I'm not too worried about double string processing - code simplicity seems more important.

> Yeah I also found it a bit sad to rip out the comment because I find them really helpful in some instances but since you said you don't mind, I didn't bother.

You're right.. But when I saw them again I felt sad too :)

Why not just do a regex - it should be really trivial. BTW we can apply it (and the newline rewriting) only if the connector is a logical replication connector, and just do the other stuff as before - it's a bit nice to have those comments actually appear in the PG logs. Just an idea, not very important.

Reply:
Like a "SanitizeForLogicalReplication" method which does the newlines + comment stripping, which only gets called when needed.

Reply:
> Why not just do a regex - it should be really trivial. BTW we can apply it (and the newline rewriting) only if the connector is a logical replication connector, and just do the other stuff as before - it's a bit nice to have those comments actually appear in the PG logs.

Now your suggestion comes pretty close to [what I already did](https://github.com/Brar/npgsql/blob/85caad804449b3ddf8c7307d042d9c656491c061/src/Npgsql/PostgresDatabaseInfo.cs#L115) (having an addComments parameter) and what feels a bit safer to me because the "parsing" for comments is done manually (and only once) instead of a regex.
Do you mind having a look?

Reply:
Sure, I guess we can do that too... It makes the comments in the queries heavier, and once again I don't really mind of doing expensive processing on these one-time queries.

I'm OK with whatever you want to do here.